apps/web/src/app/[locale]/[username]/_components/dashboard/in-progress-tab.tsx

```diff
@@ -1,39 +1,55 @@
 import {
   Table,
   TableBody,
-  TableCaption,
   TableCell,
   TableHead,
   TableHeader,
   TableRow,
 } from '@repo/ui/components/table';
+import { prisma } from '@repo/db';
+import { cache } from 'react';
+import Link from 'next/link';
+import { getRelativeTime } from '~/utils/relativeTime';
 
-const challengesList = [
-  {
-    challengeTitle: 'Extreme Challenge 1',
-    challengeStatus: 'In-Progress',
-  },
-  {
-    challengeTitle: 'Extreme Challenge 4',
-    challengeStatus: 'In-Progress',
-  },
-];
+const getInProgressSubmissions = cache(async (id: string) => {
+  return await prisma.submission.findMany({
+    where: {
+      userId: id,
+      isSuccessful: false,
+    },
+    orderBy: [
+      {
+        createdAt: 'desc',
+      },
+    ],
+    take: 25,
+    include: {
+      challenge: true,
+    },
+    distinct: 'challengeId',
+  });
+});
+
+export async function InProgressTab({ userId }: { userId: string }) {
+  const submissions = await getInProgressSubmissions(userId);
 
-export function InProgressTab() {
   return (
     <Table>
-      <TableCaption>A list of your recent challenges.</TableCaption>
       <TableHeader>
         <TableRow>
           <TableHead>Challenge</TableHead>
-          <TableHead>Status</TableHead>
+          <TableHead>Last Submission</TableHead>
         </TableRow>
       </TableHeader>
       <TableBody>
-        {challengesList.map((challengeTitle) => (
-          <TableRow key={challengeTitle.challengeTitle}>
-            <TableCell className="font-medium">{challengeTitle.challengeTitle}</TableCell>
-            <TableCell>{challengeTitle.challengeStatus}</TableCell>
+        {submissions.map((submission) => (
+          <TableRow key={submission.id}>
+            <TableCell className="font-medium underline">
+              <Link href={`/challenge/${submission.challenge.id}`}>
+                {submission.challenge.name}
+              </Link>
+            </TableCell>
+            <TableCell>{getRelativeTime(submission.createdAt)}</TableCell>
```

Comment:
As you can see, you're iterating through `submissions` but you're technically listing out `challenges` instead, so I definitely think it is a good idea to get a list of challenges instead as suggested above!